### PR TITLE
add Zonotope enclosure of SPZ

### PR DESCRIPTION
**Pull request by lucaferranti**
add Zonotope enclosure of SPZ

### CODE REVIEW

File: src/Approximations/overapproximate.jl [hunk 1]
```diff
@@ -714,6 +714,22 @@ end
 # Overapproximation of Sparse Polynomial Zonotopes
 # ================================================
 
+#=
+Given center, (dependent) generator matrix and exponent matrix of a (simple) sparse polynomialzonotope,
```
Comment (schillic) [inline src/Approximations/overapproximate.jl:718]:
```suggestion
Given center, (dependent) generator matrix and exponent matrix of a (simple) sparse polynomial zonotope,
```

File: src/Approximations/overapproximate.jl [hunk 2]
```diff
@@ -714,6 +714,22 @@ end
...
+compute thew new center and generator matrix of its zonotope overapproximation.
+=#
+function _zonotope_overapprox(c, G, E)
+    cnew = copy(c)
+    Gnew = copy(G)
```
Comment (schillic) [inline src/Approximations/overapproximate.jl:723]: I suggest to use an in-place method and pass copies to it, like this:
```suggestion
function _zonotope_overapprox!(c, G, E)
```

File: src/Approximations/overapproximate.jl [hunk 3]
```diff
@@ -714,6 +714,22 @@ end
...
+    @inbounds for (j, g) in enumerate(eachcol(G))
+        if all(iseven, E[:, j])
+            cnew .+= 0.5 * g
+            Gnew[:, j] ./= 2
```
Comment (schillic) [inline src/Approximations/overapproximate.jl:727]: This is somewhat inconsistent. 
```suggestion
            cnew .+= g / 2
            Gnew[:, j] ./= 2
```

**Review by schillic — APPROVED** [review]

File: src/Approximations/overapproximate.jl (hunk 2 above)
Comment (lucaferranti) [reply src/Approximations/overapproximate.jl:723]: do you mean to have both in-place and non-inplace helpers or just in-place helper and call `copy` in the exported `overapproximate` ?

Comment (schillic) [reply src/Approximations/overapproximate.jl:723]: Both are fine, but the latter is sufficient. It is only a suggestion, though.

Comment (lucaferranti) [reply src/Approximations/overapproximate.jl:723]: looking at how the `overapproximate` is written, maybe having both `_zonotope_overapprox!` and `_zonotope_overapprox` might be clearer?

**Review by schillic — APPROVED** [review]

**Review by mforets — APPROVED** [review]

**Pull request merged**